mesheryctl/internal/cli/root/system/connections.go

```diff
@@ -0,0 +1,304 @@
+		defer resp.Body.Close()
+
+		data, err := io.ReadAll(resp.Body)
+		if err != nil {
+			utils.Log.Error(err)
+			return err
+		}
+
+		connectionsResponse := &connections.ConnectionPage{}
+		err = json.Unmarshal(data, connectionsResponse)
+		if err != nil {
+			utils.Log.Error(err)
+			return err
+		}
+
+		header := []string{"id", "Name", "Type", "Status"}
+		rows := [][]string{}
+
+		for _, connection := range connectionsResponse.Connections {
+			if len(connection.Name) > 0 {
+				rows = append(rows, []string{connection.ID.String(), connection.Name, connection.Type, string(connection.Status)})
+			}
+		}
+
+		if len(rows) == 0 {
+			// if no connection is found
+			whiteBoardPrinter.Println("No connection(s) found")
+			return nil
+		}
+
+		if cmd.Flags().Changed("page") {
+			utils.PrintToTable(header, rows)
+		} else {
+			startIndex := 0
+			endIndex := min(len(rows), startIndex+maxRowsPerPage)
+			for {
+				// Clear the entire terminal screen
+				utils.ClearLine()
+
+				// Print number of connections and current page number
+				whiteBoardPrinter.Print("Total number of models: ", len(rows))
+				fmt.Println()
+				whiteBoardPrinter.Print("Page: ", startIndex/maxRowsPerPage+1)
+				fmt.Println()
+
+				whiteBoardPrinter.Println("Press Enter or ↓ to continue, Esc or Ctrl+C (Ctrl+Cmd for OS user) to exit")
+
+				utils.PrintToTable(header, rows[startIndex:endIndex])
+				keysEvents, err := keyboard.GetKeys(10)
+				if err != nil {
+					return err
+				}
+
+				defer func() {
+					_ = keyboard.Close()
+				}()
+
+				event := <-keysEvents
+				if event.Err != nil {
+					utils.Log.Error(fmt.Errorf("unable to capture keyboard events"))
+					break
+				}
+
+				if event.Key == keyboard.KeyEsc || event.Key == keyboard.KeyCtrlC {
+					break
+				}
+
+				if event.Key == keyboard.KeyEnter || event.Key == keyboard.KeyArrowDown {
+					startIndex += maxRowsPerPage
+					endIndex = min(len(rows), startIndex+maxRowsPerPage)
+				}
+
+				if startIndex >= len(rows) {
+					break
+				}
+			}
+		}
+
+		return nil
+	},
+}
+
+var deleteConnectionCmd = &cobra.Command{
+	Use:   "delete",
+	Short: "Delete a connection",
+	Long: `Delete
+a connection`,
+
+	Example: `
+// Delete a connection
+mesheryctl system connections delete <connection_id>
+`,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		//Check prerequisite
+
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			return err
+		}
+		err = utils.IsServerRunning(mctlCfg.GetBaseMesheryURL())
+		if err != nil {
+			return err
+		}
+		ctx, err := mctlCfg.GetCurrentContext()
+		if err != nil {
+			return err
+		}
+		err = ctx.ValidateVersion()
+		if err != nil {
+			return err
+		}
+		return nil
+	},
+
+	RunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) != 1 {
+			return errors.New(utils.SystemModelSubError("this command takes exactly one argument\n", "delete"))
```

Comment:
Use meshkit error, and be specific about the argument expected.
Can we show an example to the user in case of error?

Reply:
cmd.Usage() ?